tests/singlefolder.py

```diff
@@ -0,0 +1,170 @@
+import os
+import shelve
+import shutil
+import threading
+import typing
+
+import audeer
+
+import audbackend
+
+
+class SingleFolder(audbackend.Backend):
+    r"""Backend implemented in a single folder.
+
+    Files put on the backend
+    are stored under a random file name.
+    A serialized dictionary
+    stores the dependency between
+    backend path and the names.
+    It also stores the version
+    and checksum for every file.
```

Comment:
Just to see if I understood it correctly, if we put `file.txt` with version `1.0.0`, modify it and put it with version `2.0.0` we will have a mapping like

| backend path | version | checksum | random file name |
| --- | --- | --- | --- |
| `file.txt` | `1.0.0` | `<checksum0>` | `adnjedilhjgrpgjop` |
| `file.txt` | `2.0.0` | `<checksum1>` | `asjidofkwkeptokmd` |

whereas the random file name is used to store the file on the backend.

Reply:
yes

Reply:
Cool, makes sense to me and should be sufficient as a proof that we support single folder backends.

Reply:
I changed the title of the PR accordingly.